core/templates/dev/head/domain/collection/GuestCollectionProgressService.js

```diff
@@ -0,0 +1,146 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service that records progress guests make during a collection
+ * playthrough. Note that this service does not currently support saving a
+ * user's progress when they create an account.
+ */
+
+// TODO(bhenning): Move this to a shared service which stores state across the
+// frontend in a way that can be persisted in the backend upon account
+// creation.
+
+// TODO(bhenning): This should be reset upon login, otherwise the progress will
+// be different depending on the user's logged in/logged out state.
+
+oppia.factory('GuestCollectionProgressService', [
+  function() {
+    var COLLECTION_STORAGE_KEY = 'collectionProgressStore';
+
+    var setCollectionProgressStore = function(store) {
+      window.localStorage[COLLECTION_STORAGE_KEY] = JSON.stringify(store);
+    };
+
+    var getCollectionProgressStore = function() {
+      var store = window.localStorage[COLLECTION_STORAGE_KEY];
+      if (!store) {
+        store = {};
+        setCollectionProgressStore(store);
+        return store;
+      }
+      return JSON.parse(store);
```

Comment:
Should we be converting the stored thing to a domain object as soon as it's retrieved, rather than just working with raw dicts?

Reply:
Seems reasonable; done.